src/commands/force/source/retrieve.ts

```diff
@@ -105,12 +110,20 @@ export class Retrieve extends SourceCommand {
     }
   }
 
-  protected formatResult(): RetrieveCommandResult {
+  protected async formatResult(): Promise<RetrieveCommandResult> {
+    const packages: PackageRetrieval[] = [];
+    const projectPath = await SfdxProject.resolveProjectPath();
+
+    this.getFlag<string[]>('packagenames', []).forEach((name) => {
+      packages.push({ name, path: join(projectPath, name) });
+    });
+
     const formatterOptions = {
       waitTime: this.getFlag<Duration>('wait').quantity,
       verbose: this.getFlag<boolean>('verbose', false),
     };
     const formatter = new RetrieveResultFormatter(this.logger, this.ux, formatterOptions, this.retrieveResult);
+    formatter.packages = packages;
```

Comment:
What do you think about changing the `formatterOptions` type to add an optional `packages` option instead?

Reply:
I was doing that at first, but that type is defined in the `ResultFormatter` class, which is a base class for `deploy` and `retrieve`, but `retrieve` only has packagenames. It felt awkward altering the base class for only one of the child classes... but I'm open to doing that way too

Reply:
Why can't we do this:
```javascript
export interface RetrieveResultFormatterOptions extends ResultFormatterOptions {
  packages?: string;
}
```

and then:
`public constructor(logger: Logger, ux: UX, options: RetrieveResultFormatterOptions, result: RetrieveResult) {`